### DISCUSSION TITLE
이벤트 훅(Web Hook) 수신 문의

### DISCUSSION THREAD

**Discussion by kimdookyoung** [commerce-api-naver/commerce-solutions#62] [묻고 답하기]
안녕하세요

이벤트 훅(Web Hook) 수신 문의 드립니다.

판매자가 솔루션 사용해지 이벤트 발생시 수신 받으려고 하는데요

솔루션 등록 페이지에서 이벤트 훅(Web Hook) 영역에 
훅 수신 API Header, 훅 수신 API Key 정보 입력하는 예시가 있을까요?
테스트를 위해 임의값을 넣고 검증하기 버튼을 누르면 

Status : 400
Body : 유효하지 않은 정보입니다. 요청 API를 다시 확인해주세요.

메세지가 나옵니다~

확인 부탁드립니다.

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

커머스API센터에서 이벤트 훅 관련 정보의 검증은 실제 HTTP Request 호출 결과값을 표시합니다.
이벤트 훅 수신 과정에서의 Request 헤더와 헤더 값은 솔루션 개발사 재량으로 대상 HTTP Request가 커머스솔루션마켓으로부터 발생한 Request임을 인지할 수 있는 임의의 정보를 입력해주시면 됩니다. (단, 보안을 위해 외부 공개는 삼가 부탁드립니다.)

입력해주신 정보를 솔루션 정보로 저장하기 이전 검증 단계에서도 입력하신 헤더 및 헤더 값 그대로 HTTP Request가 발송됩니다.
해당 검증용 Request에서 [솔루션 개발 가이드](https://apicenter.commerce.naver.com/ko/basic/solution-doc#section/%EA%B0%9C%EB%B0%9C%ED%95%98%EA%B8%B0/%EC%9D%B4%EB%B2%A4%ED%8A%B8-%ED%9B%85-%EC%97%B0%EB%8F%99-%EC%9A%94%EC%86%8C-%EA%B0%80%EC%9D%B4%EB%93%9C)에 제시된 것과 같이 200 OK Status를 응답할 수 있어야 합니다.

말씀해주신 내용만으로 볼 때 입력하신 URL과 헤더/헤더 값으로 HTTP Request 발생 시 400 Bad request status를 반환하는 상황일 수도 있을 것 같습니다.
입력해주신 URL로 HTTP Request가 정상적으로 처리되는지 확인해보시기 바랍니다.

**Reply by kimdookyoung** [reply]
죄송하지만, 임의값 넣고 테스트시 '요청이 실패하였습니다. 다시 한 번 시도해주세요.'라고 나오는건 입력된 값들이 잘못되어서 그런걸까요??

**Reply by commerce-api-naver** [reply]
@kimdookyoung 님, 커머스API 이민석입니다.

해당 메시지라면 입력하신 '훅 수신 API Path' URL로 정상적인 HTTP Request 발송에 실패하였을 확률이 높습니다.
혹시 입력하신 대상 URL이 공개된 인터넷 망에서 접근 가능한 URL이었을까요?

**Reply by kimdookyoung** [reply]
방화벽 이슈로 확인됩니다~ 확인 감사합니다~
